### PR TITLE
Small improvement to rotate_y::hit() function

### LINKED ISSUE #1484: [Book 2, Chapter 8.2] Improve code readability & efficiency
In Listing 69, function `hit()` of the `rotate_y` instance, origin and direction of the ray are copied into local variables and then _x_ and _z_ values are promptly overwritten. This is not only inefficient, but also requires us to use hard-coded axis numbers (instead of referring to them as _x_, _y_, and _z_), which can be error-prone.

The code can be improved as follows:

```diff
     bool hit(const ray& r, interval ray_t, hit_record& rec) const override {
         // Change the ray from world space to object space
-        auto origin = r.origin();
-        auto direction = r.direction();
-
-        origin[0] = cos_theta*r.origin()[0] - sin_theta*r.origin()[2];
-        origin[2] = sin_theta*r.origin()[0] + cos_theta*r.origin()[2];
-
-        direction[0] = cos_theta*r.direction()[0] - sin_theta*r.direction()[2];
-        direction[2] = sin_theta*r.direction()[0] + cos_theta*r.direction()[2];
+        point3 origin {
+            cos_theta*r.origin().x() - sin_theta*r.origin().z(),
+            r.origin().y(),
+            sin_theta*r.origin().x() + cos_theta*r.origin().z()
+        };
+
+        vec3 direction {
+            cos_theta*r.direction().x() - sin_theta*r.direction().z(),
+            r.direction().y(),
+            sin_theta*r.direction().x() + cos_theta*r.direction().z()
+        };
 
         ray rotated_r(origin, direction, r.time());
```

Same this further down in the function:

```diff
         // Change the intersection point from object space to world space
-        auto p = rec.p;
-        p[0] =  cos_theta*rec.p[0] + sin_theta*rec.p[2];
-        p[2] = -sin_theta*rec.p[0] + cos_theta*rec.p[2];
+        point3 p {
+             cos_theta*rec.p.x() + sin_theta*rec.p.z(),
+             rec.p.y(),
+            -sin_theta*rec.p.x() + cos_theta*rec.p.z()
+        };
 
         // Change the normal from object space to world space
-        auto normal = rec.normal;
-        normal[0] =  cos_theta*rec.normal[0] + sin_theta*rec.normal[2];
-        normal[2] = -sin_theta*rec.normal[0] + cos_theta*rec.normal[2];
+        vec3 normal {
+             cos_theta*rec.normal.x() + sin_theta*rec.normal.z(),
+             rec.normal.y(),
+            -sin_theta*rec.normal.x() + cos_theta*rec.normal.z()
+        };
 
         rec.p = p;
         rec.normal = normal;
```

### PR DESCRIPTION
**Pull request by hollasch**
Resolves #1484

### CODE REVIEW

**Comment by hollasch** [comment]
@dimitry-ishenko

File: books/RayTracingTheNextWeek.html [hunk 1]
```diff
@@ -3815,34 +3815,41 @@
...
+            // Determine whether an intersection exists in object space (and if so, where).
+
             if (!object->hit(rotated_r, ray_t, rec))
                 return false;
 
-            // Change the intersection point from object space to world space
-            auto p = rec.p;
-            p[0] =  cos_theta*rec.p[0] + sin_theta*rec.p[2];
-            p[2] = -sin_theta*rec.p[0] + cos_theta*rec.p[2];
+            // Transform the intersection from object space back to world space.
```
Comment (whydoubt) [inline books/RayTracingTheNextWeek.html:3841]: Before, there was a comment about the intersection and one about the normal.  Now the one about the normal is gone. Should that just be integrated into this comment, or did you drop mention of the normal on purpose?

Comment (hollasch) [reply books/RayTracingTheNextWeek.html:3841]: I considered the normal to be part of the intersection data (which I why I didn't say "intersection _point_").

Comment (hollasch) [reply books/RayTracingTheNextWeek.html:3841]: Thanks for checking.

**Comment by hollasch** [comment]
@whydoubt — sorry about jumping the gun on this PR. Seemed like everyone else has been busy, and I'm wrapping up for a ~Sunday~  September 1 patch release (v4.0.1).

**Pull request merged**